### DISCUSSION TITLE
How should we communicate?

### DISCUSSION THREAD

**Discussion by mattjbr123** [NERC-CEH/rse_group#16] [General]
https://ben.balter.com/2020/08/14/tools-of-the-trade/

Lots of good ideas in this post that Robin shared earlier, what are folks thoughts on any of this? 
I can see it being a useful guidance document for where we could put different elements of communications.

**Comment by metazool** [comment]
This piece really chimed with me, especially "Default to open... optimise for discoverability... err on the side of over-communication"

If there's a missing piece for me, it's to communicate in code more; unfinished pull requests are one of my favourite forms of communication. Apart from the reproducible open science angle, it's another incentive to nudge more of our research collaborations to default to open. I suspect I have no idea how much there is to be learned from the rest of the RSE group...

**Reply by metazool** [reply]
Another possible item for the mix here is the Fediverse, where a lot of open source and research communication has moved or is moving. SURF in the Netherlands is running a pilot to [provide and support hosted Mastodon services](https://www.surf.nl/en/about-the-mastodon-pilot) for research institutions. I often see relevant links through the Fediverse either on software development culture or computational environmental science, which i end up posting to work channels...

The [Newcastle RSE team is in the Fediverse](https://mastodon.social/@ncl_rse) but don't seem so active. See also the [SFO Museum's custom mastodon server](https://millsfield.sfomuseum.org/blog/2024/03/12/activitypub/) for the artefacts in their collection, there are nice unexpected ways to use the ActivityPub protocol

**Comment by jmarshrossney** [comment]
I like this a lot. 

That said, there are some issues with "default to openness" that may not apply to a typical team of developers working on a shared project, but which do apply to us, a group of RSEs who individually go out and work with 'external' teams, as I mentioned [here](https://github.com/NERC-CEH/rse_group/discussions/11#discussioncomment-9869101).

Imo the Fediverse is something to keep in mind down the line but to me it makes sense to stick with the more familiar and established tools while we're getting set up.

**Comment by mattjbr123** [comment]
If nothing else this article has made me think 'oooh could this be more open/shared with more people' when helping people over email. I'm thinking I might start gently nudging such conversations away from email (to exactly what is the question, but we'll get there)

**Comment by longr** [comment]
For an example, I am on the Research Data Cloud project.

This started with a teams group to collect documents in, and will probably stay like that, but I think the RSE side could be split off to GitHub.   
I have been taking minutes and notes in them meetings that I now need to merge together as an understanding of how the project and application works.  I need the users / original builders of the apps to comment to make sure my understanding is correct.  I guess this would make sense to store and do in github?  A new repository for the project, perhaps a separate branch for the notes side?  or even notes in the main repo and delete/archive when not needed anymore?  

If done in GitHub, and all team has access, then it keeps notes with the project and allows everyone to know what is going on?  Also potentially increases the bus factor from its current level of 1.

**Comment by metazool** [comment]
It makes sense to me @longr - I would default to keeping work in this shared repository and running it by the rest of us in the form of a pull request. 

I added some notes this way soon after I started - a [holding page for notes on pipelines and frameworks](https://github.com/NERC-CEH/rse_group/tree/main/infrastructure/pipelines) including a [paper summary](https://github.com/NERC-CEH/rse_group/blob/main/infrastructure/pipelines/pangeo_forge.md) for `pangeo-forge` because it looked so appropriate for earth observation / cloud data use cases.

Didn't do this with any idea of introducing a structure, more an idea to use the repo as an alternative to a Confluence-style wiki for shared notes. Useful to manage the bus factor, as you say, and for future onboarding...

**Comment by metazool** [comment]
Comment probably belongs to another thread: just tried creating a Github `team` that could then be used to add everyone as project contributors for PR review purposes. You have to be an organisation member to enlist as a team member - at present Matts are, but @longr @jmarshrossney and @albags aren't yet. I made an email support ticket to do it for myself, came through quickly.  

https://github.com/orgs/NERC-CEH/teams/eds-rse/members

**Comment by metazool** [comment]
https://chelseatroy.com/2018/03/29/why-do-remote-meetings-suck-so-much/ - speaking as a person with a subjective caucus score of 2, this series of posts and its advice on remote meeting moderation is interesting for the standup and future workshops

**Reply by matthewcoole** [reply]
Really interesting read. I've definitely noticed a lot of the problems mentioned in the article and they're clearly exacerbated in remote meetings (I think my caucus score was 5-6). 

Perhaps we could discuss (maybe as suggestions for our team agreement) specific ways we as RSEs will run any meetings that we are involved with e.g. round room discussion, single facilitator (or possibly we need to write a teams plugin for a digital conch?).

**Reply by mattjbr123** [reply]
Yup, agreed. That was really insightful and enlightening to me - now I have a collective name for all those little problems and worries I often have during meetings! I like the moderation technique suggested, perhaps we could trial it at future meetings
